compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt

```diff
@@ -17,6 +17,7 @@ import kotlin.math.roundToInt
 fun MapViewCamera.getZoom(): Double {
   return when (this.state) {
     is CameraState.Centered -> this.state.zoom
+    is CameraState.BoundingBox -> TODO("This is gonna be a problem...")
```

Comment:
This file raises an interesting problem... While the camera will never remain in the bbox state for long (as currently implemented), we may need to make these methods failable (optional return values / silently ignoring updates). The whole point of a bbox abstraction is that you don't know the zoom and several other parameters until you compute it with a concrete map object.

Reply:
I would argue you should still be able to get zoom. On set zoom I could see it as reasonable behavior where a user:

1. Sets the camera to a bounding box.
2. Taps zoom in/zoom out and expects the camera to move just like centered.

In this case, we'd probably have to do the following for each step:

1. Move the camera to the bounding box state.
2. Convert the bbox camera to a centered state at the current "view port", then zoom in. 

Basically this lets us abandon the BBox restriction, while still obeying the user's goal of zooming in after centering on the BBox.

This same general logic flow seems to make sense for all of these methods. The only trick is we have to swap the camera to centered to allow the behavior.

Reply:
Any state except the tracking ones will end up back at centered on both iOS and Android given how our observability works. This happens automatically and immediately after the animation completes.

The only point at which this could fail is in the interim if the user ninja taps the explicit zoom button to try to increment or something. At this point in time, the camera state value that we store cannot possibly have a zoom in the case of the bbox state. This would behave unexpectedly in the first case too by the way, but it would not crash the app.

Hence I think we probably just need to make this method return a `Double?`.

Reply:
Yep. That'll work. I was overthinking per the fact that the MapView technically knows zoom even when we're not using it. But since we can reconstruct a bounding box repeatably, a `Double?` solves the problem I was thinking about.